### ISSUE TITLE
Media queries not deterministic.

### ISSUE THREAD

**Issue by Silverwolf90** [jxnblk/cxs#14]
I'm experiencing a lot of issues with media queries and out-of-order/conflicting rules. 

Does cxs handle media queries intelligently, ie: are the rules generated in the order provided by the properties? I'm defining these media queries from smallest to largest.

Sometimes it seems adding a property to a media query ends up affecting unrelated things. I haven't been able to pinpoint a specific case, it all feels very arbitrary (although I'm sure there's a good reason as to why it's happening).

**Comment by Silverwolf90** [comment]
I'm not totally sure, but this might not have anything to do with media queries. I suspect it's just noticeable because you're expecting style toggle on/off 

Commenting out all the common declarations in `/dist` seemed to fix a lot of issues I was having.

**Comment by jxnblk** [comment]
I'm not sure I follow. This is still a WIP so there's probably issues with what you're describing. Can you provide a link to a jsbin as an example?

**Comment by Silverwolf90** [comment]
Yeah I will try and get an isolated test case for you.

**Comment by Silverwolf90** [comment]
But to simplify my initial ramblings, is the order of the properties and media queries preserved? This is particularly important for media queries where you would want properties to cascade as you increase screen size.

**Comment by jxnblk** [comment]
They should be and I intended for that to be true, but there is some sorting that cxs does to rulesets that might be affecting that. I should add some tests and ensure it works as expected

**Comment by jxnblk** [comment]
Just published 1.0.0 with some small fixes. Feel free to reopen this issue if you're still having problems with the latest version

**Comment by Silverwolf90** [comment]
Still having issues. I may have isolated... some bizarre behavior? Not entirely sure if it's related.

The following block:

```javascript
const sx = {
  '@media screen and (min-width : 320px)': {
    display: 'none',
  },
  '@media screen and (min-width : 768px)': {
    display: 'block',
  },
};
```

Results in the following `cxs.rules`:

```javascript
[
  {
    id: 'cxs-1266547199',
    order: 1,
    selector: '.cxs-1266547199',
    css: '.cxs-1266547199{}'
  },
  {
    id: 'cxs-1266547199@media screen and (min-width : 320px)',
    order: 2,
    selector: '.cxs-1266547199',
    css: '@media screen and (min-width : 320px) { .cxs-1266547199{} }'
  },
  {
    id: 'cxs-1266547199@media screen and (min-width : 768px)',
    order: 2,
    selector: '.cxs-1266547199',
    css: '@media screen and (min-width : 768px) { .cxs-1266547199{} }'
  }
]
```

Which results in the following `cxs.css`:

```
.cxs-1266547199{}@media screen and (min-width : 320px) { .cxs-1266547199{} }@media screen and (min-width : 768px) { .cxs-1266547199{} }
```

So it seems to be ripping out the `display` properties?

**Comment by asilvas** [comment]
@Silverwolf90 was this resolved?

**Comment by gingur** [comment]
looks like this is resolved in 2.1.0, at least for me; was seeing the issue described above previously.
